AwsCryptographicMaterialProviders/dafny/AwsCryptographicMaterialProviders/src/CMMs/RequiredEncryptionContextCMM.dfy

```diff
@@ -52,16 +53,7 @@ module RequiredEncryptionContextCMM {
       ensures Modifies == { History } + underlyingCMM.Modifies
     {
       var keySet := inputKeys;
-      var keySeq := [];
-      while keySet != {}
-        invariant |keySeq| + |keySet| == |inputKeys|
-        invariant forall k <- keySeq
-                    :: k in inputKeys
-      {
-        var key :| key in keySet;
-        keySeq := keySeq + [key];
-        keySet := keySet - {key};
-      }
+      var keySeq := SortedSets.ComputeSetToSequence(keySet);
```

Comment:
Similarly here, I think that we could see the diff more clearly as invariant by writing it this way.
(I checked the ensures clause)
This way the `invariant` changes to an assert.

```dafny
var keySeq := SortedSets.ComputeSetToSequence(inputKeys);
assert |keySeq| == |inputKeys|;
assert forall k <- keySeq :: k in inputKeys;
```